spec/domain/command_handlers/game_command_handler_spec.rb

```diff
@@ -116,231 +137,335 @@
+          InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(60)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(70)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(90)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(110)),
+          GameRoundFinishedPlayingEvent.new(id, GameRound.new(1)),
+          GameEndedEvent.new(id, GameState::ENDED)
+        )
       end
     end
   end
 
   describe JoinGameCommand do
-    let(:command) { join_game_command }
-
-    it_behaves_like "an event publisher" do
-      before { dispatch(create_game_command) }
+    subject(:command) { JoinGameCommand.new(id: id, player_id: steven) }
+    before { given_events GameCreatedEvent.new(id, GameState::LOBBYING, bob) }
 
-      let(:expected_events) { [PlayerJoinedGameEvent.new(id, steven)] }
+    it 'joins a game' do
+      when_command(command)
+      then_events(PlayerJoinedGameEvent.new(id, steven))
     end
 
-    context "when already in game" do
-      before do
-        dispatch(create_game_command)
-        dispatch(join_game_command)
-      end
+    context 'given the player already joined the game' do
+      before { given_events PlayerJoinedGameEvent.new(id, steven) }
 
-      specify do
-        expect { dispatch(command.dup) }.to raise_error(PlayerAlreadyInGameError)
+      it 'denies joining the game again' do
+        expect { dispatch(command) }.to raise_error(PlayerAlreadyInGameError)
       end
     end
 
-    context "when game already started" do
-      before do
-        dispatch(create_game_command)
-        dispatch(start_game_command)
-      end
+    context 'given the game already started' do
+      before { given_events GameStartedEvent.new(id, GameState::RUNNING, deck, board) }
 
-      specify do
+      it 'denies joining the game' do
         expect { dispatch(command) }.to raise_error(GameAlreadyStartedError)
       end
     end
   end
 
   describe LeaveGameCommand do
-    let(:command) { leave_game_command }
-
-    it_behaves_like "an event publisher" do
-      before do
-        dispatch(create_game_command)
-        dispatch(join_game_command)
-      end
+    subject(:command) { LeaveGameCommand.new(id: id, player_id: steven) }
+    before { given_events GameCreatedEvent.new(id, GameState::LOBBYING, bob) }
 
-      let(:expected_events) { [PlayerLeftGameEvent.new(id, steven)] }
-    end
+    context 'given the player joined the game' do
+      before { given_events PlayerJoinedGameEvent.new(id, steven) }
 
-    context "when not in game" do
-      before do
-        dispatch(create_game_command)
+      it 'leaves a game' do
+        when_command(command)
+        then_events(PlayerLeftGameEvent.new(id, steven))
       end
 
-      specify do
-        expect { dispatch(command) }.to raise_error(PlayerNotInGameError)
-      end
-    end
+      context 'and the game already started' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.